SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -110,11 +110,18 @@ void ComponentReader::ApplyComponentData(const Worker_ComponentId ComponentId, S
 	Schema_Object* ComponentObject = Schema_GetComponentDataFields(Data);
 
 	// ComponentData will be missing fields if they are completely empty (options, lists, and maps).
-	// However, we still want to apply this empty data, so we need the full list of field IDs for
-	// that component type (Data, OwnerOnly, Handover, etc.).
-	const TArray<Schema_FieldId>& InitialIds = ClassInfoManager->GetFieldIdsByComponentId(ComponentId);
+	// Additional they may be empty if they are currently not active and not being replicated.
+	TArray<Schema_FieldId> UpdatedIds;
+	UpdatedIds.SetNumUninitialized(Schema_GetUniqueFieldIdCount(ComponentObject));
```

Comment:
Could reserve space for both up front.

Reply:
The id sets overlap so it would be slightly wasteful